--- a/src/sqlite-vss.cpp	2023-08-06 04:25:15.000000000 +0200
+++ b/src/sqlite-vss.cpp	2025-02-17 16:08:41.428840481 +0100
@@ -15,6 +15,7 @@
 #include <faiss/impl/AuxIndexStructures.h>
 #include <faiss/impl/IDSelector.h>
 #include <faiss/impl/io.h>
+#include <faiss/impl/FaissException.h>
 #include <faiss/index_factory.h>
 #include <faiss/index_io.h>
 #include <faiss/utils/distances.h>
@@ -182,7 +183,7 @@
                              -1);
         return;
     }
-    
+
     float inner_product = faiss::fvec_inner_product(lhs->data(), rhs->data(), lhs->size());
     float lhs_norm = faiss::fvec_norm_L2sqr(lhs->data(), lhs->size());
     float rhs_norm = faiss::fvec_norm_L2sqr(rhs->data(), rhs->size());
