--- old/src/engine/process.cpp	2015-01-16 16:31:33.000000000 +0100
+++ new/src/engine/process.cpp	2015-02-04 23:33:57.370601537 +0100
@@ -290,7 +290,7 @@
 			}
 
 			// Execute process
-			execl(cmd, args, (char*)0); // noreturn on success
+			execl(cmd.c_str(), args.c_str(), (char*)0); // noreturn on success
 
 			_exit(-1);
 		}
--- old/src/engine/sftpcontrolsocket.cpp	2014-12-14 05:30:09.000000000 +0100
+++ new/src/engine/sftpcontrolsocket.cpp	2015-02-04 23:36:03.740600605 +0100
@@ -1091,7 +1091,7 @@
 		return FZ_REPLY_INTERNALERROR;
 	}
 
-	pData->pParser->AddLine(entry);
+	pData->pParser->AddLine(entry.c_str());
 
 	return FZ_REPLY_WOULDBLOCK;
 }
