From fdef654288eb37a7c999ae71b216fe292cba8217 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sun, 21 Aug 2022 20:27:02 +0200
Subject: [PATCH] fix minizip util

---
 Source/Core/Common/MinizipUtil.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/Core/Common/MinizipUtil.h b/Source/Core/Common/MinizipUtil.h
index 6057632ddd..5f56053405 100644
--- a/Source/Core/Common/MinizipUtil.h
+++ b/Source/Core/Common/MinizipUtil.h
@@ -5,7 +5,8 @@
 
 #include <algorithm>
 
-#include <mz_compat.h>
+#include <minizip/zip.h>
+#include <minizip/unzip.h>
 
 #include "Common/CommonTypes.h"
 #include "Common/ScopeGuard.h"
-- 
2.37.2

