From 3bd8b7ee130fb257fe983ccb3ccbc772c50e73ca Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Thu, 26 Sep 2024 17:35:38 +0200
Subject: [PATCH] fix disabled achievements

---
 Source/Core/DolphinTool/VerifyCommand.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Source/Core/DolphinTool/VerifyCommand.cpp b/Source/Core/DolphinTool/VerifyCommand.cpp
index b80f00a273..366884134b 100644
--- a/Source/Core/DolphinTool/VerifyCommand.cpp
+++ b/Source/Core/DolphinTool/VerifyCommand.cpp
@@ -163,11 +163,13 @@ int VerifyCommand(const std::vector<std::string>& args)
   verifier.Finish();
   const DiscIO::VolumeVerifier::Result& result = verifier.GetResult();
 
+#ifdef USE_RETRO_ACHIEVEMENTS
   // Calculate rcheevos hash
   if (rc_hash_calculate)
   {
     rc_hash_result = AchievementManager::CalculateHash(input_file_path);
   }
+#endif
 
   // Print the report
   if (!algorithm_is_set)
-- 
2.46.1

