From d3708b30740563fd2babeb7f651083de8704ff72 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sun, 21 Aug 2022 20:11:31 +0200
Subject: [PATCH] system libspng

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6daf29220..5338f39c24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -811,7 +811,7 @@ else()
   set(LZO lzo2)
 endif()
 
-add_subdirectory(Externals/libspng)
+pkg_check_modules(spng libspng)
 
 # Using static FreeSurround from Externals
 # There is no system FreeSurround library.
-- 
2.37.2

