From d98f109113cd16d640ed57cf29c009e2d360d8aa Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sat, 25 Feb 2023 09:54:13 +0100
Subject: [PATCH] system libspng

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2530c1a0ef..1bdb3d03d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,7 +803,7 @@ else()
   set(LZO lzo2)
 endif()
 
-pkg_check_modules(pc_spng IMPORTED_TARGET spng)
+pkg_check_modules(pc_spng IMPORTED_TARGET libspng)
 if (pc_spng_FOUND AND TARGET PkgConfig::pc_spng)
   message(STATUS "Using the system libspng")
   set(spng_target PkgConfig::pc_spng)
-- 
2.39.2

