From c5d4419a34e8bd5593d777992a2035995a753332 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Mon, 7 Aug 2023 16:16:46 +0200
Subject: [PATCH] more system libs

---
 CMakeLists.txt | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee44d04458..0c8bad9eb3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -678,11 +678,11 @@ dolphin_find_optional_system_library_pkgconfig(ZSTD libzstd>=1.4.0 zstd::zstd Ex
 
 dolphin_find_optional_system_library_pkgconfig(ZLIB zlib-ng ZLIB::ZLIB Externals/zlib-ng)
 
-dolphin_find_optional_system_library_pkgconfig(MINIZIP minizip>=3.0.0 minizip::minizip Externals/minizip)
+dolphin_find_optional_system_library_pkgconfig(MINIZIP minizip-ng>=3.0.0 minizip::minizip Externals/minizip)
 
 dolphin_find_optional_system_library(LZO Externals/LZO)
 
-dolphin_find_optional_system_library_pkgconfig(SPNG spng spng::spng Externals/libspng)
+dolphin_find_optional_system_library_pkgconfig(SPNG libspng spng::spng Externals/libspng)
 
 # Using static FreeSurround from Externals
 # There is no system FreeSurround library.
-- 
2.41.0

