diff --git a/Source/Core/Core/IOS/Network/Socket.cpp b/Source/Core/Core/IOS/Network/Socket.cpp
index d7464acc4f..a9471de5ca 100644
--- a/Source/Core/Core/IOS/Network/Socket.cpp
+++ b/Source/Core/Core/IOS/Network/Socket.cpp
@@ -7,6 +7,8 @@
 
 #include <algorithm>
 #include <mbedtls/error.h>
+#include <mbedtls/x509_crt.h>
+
 #ifndef _WIN32
 #include <arpa/inet.h>
 #include <unistd.h>
@@ -381,7 +383,7 @@ void WiiSocket::Update(bool read, bool write, bool except)
             // Below is an alternative to dump the peer certificate
             if (SConfig::GetInstance().m_SSLDumpPeerCert && ctx->session_negotiate != nullptr)
             {
-              const mbedtls_x509_crt* cert = ctx->session_negotiate->peer_cert;
+              const mbedtls_x509_crt* cert = mbedtls_ssl_get_peer_cert(ctx);
               if (cert != nullptr)
               {
                 std::string filename = File::GetUserPath(D_DUMPSSL_IDX) +
diff --git a/Source/Core/Core/CMakeLists.txt b/Source/Core/Core/CMakeLists.txt
index 2b7c141e68..764c0654f1 100644
--- a/Source/Core/Core/CMakeLists.txt
+++ b/Source/Core/Core/CMakeLists.txt
@@ -541,6 +541,8 @@ elseif(_M_ARM_64)
   )
 endif()
 
+target_compile_definitions(core PRIVATE MBEDTLS_X509_CRT_PARSE_C MBEDTLS_SSL_CLI_C MBEDTLS_SSL_RENEGOTIATION)
+
 target_link_libraries(core
 PUBLIC
   audiocommon
