diff --git a/Source/Core/Common/SettingsHandler.cpp b/Source/Core/Common/SettingsHandler.cpp
index e7b3d5d066..23c64f1e2a 100644
--- a/Source/Core/Common/SettingsHandler.cpp
+++ b/Source/Core/Common/SettingsHandler.cpp
@@ -11,7 +11,7 @@
 #include <iomanip>
 #include <string>
 
-#include <fmt/time.h>
+#include <fmt/chrono.h>
 
 #include "Common/CommonTypes.h"
 
diff --git a/Source/Core/Core/Core.cpp b/Source/Core/Core/Core.cpp
index 5908cc1347..579fa5727a 100644
--- a/Source/Core/Core/Core.cpp
+++ b/Source/Core/Core/Core.cpp
@@ -13,7 +13,7 @@
 #include <variant>
 
 #include <fmt/format.h>
-#include <fmt/time.h>
+#include <fmt/chrono.h>
 
 #ifdef _WIN32
 #include <windows.h>
