diff --git a/Source/Core/UICommon/AutoUpdate.cpp b/Source/Core/UICommon/AutoUpdate.cpp
index 003135683a..339bac64a6 100644
--- a/Source/Core/UICommon/AutoUpdate.cpp
+++ b/Source/Core/UICommon/AutoUpdate.cpp
@@ -4,7 +4,7 @@
 
 #include "UICommon/AutoUpdate.h"
 
-#include <picojson/picojson.h>
+#include <picojson.h>
 #include <string>
 
 #include "Common/CommonPaths.h"
diff --git a/Source/Core/UICommon/ResourcePack/Manifest.cpp b/Source/Core/UICommon/ResourcePack/Manifest.cpp
index 37101b2639..529911ab2c 100644
--- a/Source/Core/UICommon/ResourcePack/Manifest.cpp
+++ b/Source/Core/UICommon/ResourcePack/Manifest.cpp
@@ -4,7 +4,7 @@
 
 #include "UICommon/ResourcePack/Manifest.h"
 
-#include <picojson/picojson.h>
+#include <picojson.h>
 
 namespace ResourcePack
 {
