diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09f43fe6d2..4e9dc9835e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,12 +28,17 @@ option(ENABLE_QT "Enable Qt (Default)" ON)
 option(ENABLE_LTO "Enables Link Time Optimization" OFF)
 option(ENABLE_GENERIC "Enables generic build that should run on any little-endian host" OFF)
 option(ENABLE_HEADLESS "Enables running Dolphin as a headless variant" OFF)
+option(USE_SHARED_GLSLANG "Use system glslang if found instead of bundled version.")
+option(USE_SHARED_FMT "Use system fmt if found instead of bundled version.")
+option(USE_SHARED_XXHASH "Use system xxhash if found instead of bundled version.")
 option(ENABLE_ALSA "Enables ALSA sound backend" ON)
 option(ENABLE_PULSEAUDIO "Enables PulseAudio sound backend" ON)
 option(ENABLE_LLVM "Enables LLVM support, for disassembly" ON)
 option(ENABLE_TESTS "Enables building the unit tests" ON)
 option(USE_DISCORD_PRESENCE "Enables Discord Rich Presence, show the current game on Discord" ON)
 
+set(BUILD_SHARED_LIBS OFF)
+
 # Maintainers: if you consider blanket disabling this for your users, please
 # consider the following points:
 #  * No data is being sent without explicit user approval (pop up box at first
@@ -539,9 +544,18 @@ else()
   message(STATUS "Using static fmt from Externals")
   add_subdirectory(Externals/fmt EXCLUDE_FROM_ALL)
 endif()
-add_subdirectory(Externals/glslang)
 add_subdirectory(Externals/imgui)
 
+if(USE_SHARED_GLSLANG)
+  set(GLSLANG_FOUND TRUE)
+endif()
+if (GLSLANG_FOUND)
+  message(STATUS "Using shared glslang")
+else()
+  message(STATUS "Using static glslang from Externals")
+  add_subdirectory(Externals/glslang)
+endif()
+
 find_package(pugixml)
 if(NOT pugixml_FOUND)
   message(STATUS "Using static pugixml from Externals")
@@ -567,7 +588,19 @@ else()
   add_subdirectory(Externals/enet)
 endif()

-if(NOT XXHASH_FOUND)
+if(USE_SHARED_XXHASH)
+  check_lib(XXHASH libxxhash xxhash xxhash.h QUIET)
+  if (XXHASH_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES ${XXHASH_INCLUDE_DIRS})
+    # hack: LDFLAGS already contains -lxxhash but all flags but the first are
+    # dropped; ugh, cmake
+    set(CMAKE_REQUIRED_FLAGS ${XXHASH_LDFLAGS})
+    set(CMAKE_REQUIRED_LIBRARIES ${XXHASH_LIBRARIES})
+  endif()
+endif()
+if(XXHASH_FOUND)
+  message(STATUS "Using shared xxhash")
+else()
   message(STATUS "Using static xxhash from Externals")
   add_subdirectory(Externals/xxhash)
 endif()
