diff --git a/src/glsl/glslang.cc b/src/glsl/glslang.cc
index 5546120..767af44 100644
--- a/src/glsl/glslang.cc
+++ b/src/glsl/glslang.cc
@@ -210,6 +210,7 @@ const TBuiltInResource DefaultTBuiltInResource = {
     /* .maxTaskWorkGroupSizeY_NV = */ 1,
     /* .maxTaskWorkGroupSizeZ_NV = */ 1,
     /* .maxMeshViewCountNV = */ 4,
+    /* .maxDualSourceDrawBuffersEXT = */ 1,
 #endif
 
     /* .limits = */ {
