--- old/src/ptclib/pssl.cxx	2014-02-17 03:14:34.000000000 +0100
+++ new/src/ptclib/pssl.cxx	2016-06-07 10:02:26.950181503 +0200
@@ -1189,9 +1189,11 @@
   SSL_METHOD * meth;
 
   switch (method) {
+#if !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2030000fL
     case SSLv3:
       meth = SSLv3_method();
       break;
+#endif
     case TLSv1:
       meth = TLSv1_method(); 
       break;
--- old/include/ptclib/pssl.h	2014-02-17 03:14:30.000000000 +0100
+++ new/include/ptclib/pssl.h	2016-06-07 10:02:09.620181631 +0200
@@ -488,7 +488,9 @@
   public:
     enum Method {
       SSLv23,
+#if !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2030000fL
       SSLv3,
+#endif
       TLSv1
     };
 
