--- a/desmume/src/wifi.cpp
+++ b/desmume/src/wifi.cpp
@@ -55,6 +55,6 @@

 // Some platforms need HAVE_REMOTE to work with libpcap, but
 // Apple platforms are not among them.
-#ifndef __APPLE__
+#ifdef __WIN32__
 	#define HAVE_REMOTE
 #endif
--- a/desmume/src/wifi.h
+++ b/desmume/src/wifi.h
@@ -29,6 +29,10 @@
 #include <string>
 #include <vector>

+#ifdef Status
+#undef Status
+#endif
+
 #define		REG_WIFI_ID					0x000
 #define     REG_WIFI_MODE       		0x004
 #define     REG_WIFI_WEP        		0x006
