--- a/setup.py	2023-11-13 21:41:18.143247550 +0100
+++ b/setup.py	2023-11-13 21:40:21.233168063 +0100
@@ -200,7 +200,8 @@
             if nvcc_flags == "":
                 nvcc_flags = []
             else:
-                nvcc_flags = nvcc_flags.split(" ")
+                import shlex
+                nvcc_flags = shlex.split(nvcc_flags)
         else:
             define_macros += [("WITH_HIP", None)]
             nvcc_flags = []
