diff --git a/libshaderc_util/include/libshaderc_util/compiler.h b/libshaderc_util/include/libshaderc_util/compiler.h
index f36a97f..58939c8 100644
--- a/libshaderc_util/include/libshaderc_util/compiler.h
+++ b/libshaderc_util/include/libshaderc_util/compiler.h
@@ -23,7 +23,7 @@
 #include <unordered_map>
 #include <utility>

-#include "glslang/Public/ShaderLang.h"
+#include "glslang/ShaderLang.h"

 #include "counting_includer.h"
 #include "file_finder.h"
diff --git a/libshaderc_util/include/libshaderc_util/counting_includer.h b/libshaderc_util/include/libshaderc_util/counting_includer.h
index 3e6b0a6..15e2e7f 100644
--- a/libshaderc_util/include/libshaderc_util/counting_includer.h
+++ b/libshaderc_util/include/libshaderc_util/counting_includer.h
@@ -17,7 +17,7 @@

 #include <atomic>

-#include "glslang/Public/ShaderLang.h"
+#include "glslang/ShaderLang.h"

 #include "libshaderc_util/mutex.h"

diff --git a/libshaderc_util/include/libshaderc_util/shader_stage.h b/libshaderc_util/include/libshaderc_util/shader_stage.h
index a939cb2..d81b964 100644
--- a/libshaderc_util/include/libshaderc_util/shader_stage.h
+++ b/libshaderc_util/include/libshaderc_util/shader_stage.h
@@ -20,7 +20,7 @@
 #include <utility>
 #include <vector>

-#include "glslang/Public/ShaderLang.h"
+#include "glslang/ShaderLang.h"

 #include "libshaderc_util/string_piece.h"

--- a/libshaderc_util/CMakeLists.txt	2018-11-14 02:00:15.570363165 +0100
+++ b/libshaderc_util/CMakeLists.txt	2018-11-14 02:00:38.239552623 +0100
@@ -35,7 +35,9 @@
 find_package(Threads)
 target_link_libraries(shaderc_util PRIVATE
   glslang OSDependent OGLCompiler glslang HLSL SPIRV
-  SPIRV-Tools-opt SPIRV-Tools ${CMAKE_THREAD_LIBS_INIT})
+  SPIRV-Tools-opt SPIRV-Tools-opt-shared
+  SPIRV-Tools-shared SPIRV-Tools
+  ${CMAKE_THREAD_LIBS_INIT})

 shaderc_add_tests(
   TEST_PREFIX shaderc_util
diff --git a/libshaderc_spvc/CMakeLists.txt b/libshaderc_spvc/CMakeLists.txt
index fbc4cf2..810754e 100644
--- a/libshaderc_spvc/CMakeLists.txt
+++ b/libshaderc_spvc/CMakeLists.txt
@@ -22,7 +22,7 @@ target_include_directories(shaderc_spvc
   ${shaderc_SOURCE_DIR}/libshaderc_util/include
   ${spirv-tools_SOURCE_DIR}/include
   ${spirv-tools_SOURCE_DIR}/
-  ${SPIRV-Cross_SOURCE_DIR}/..
+  ${SPIRV-Cross_SOURCE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}/../third_party/spirv-tools/)
 
 if(SHADERC_ENABLE_SPVC_PARSER)
