diff --git a/libshaderc_spvc/src/spvc.cc b/libshaderc_spvc/src/spvc.cc
index 8ecfd9a..b5427f9 100644
--- a/libshaderc_spvc/src/spvc.cc
+++ b/libshaderc_spvc/src/spvc.cc
@@ -15,9 +15,9 @@
 #include "shaderc/spvc.h"
 #include "libshaderc_util/exceptions.h"
 
-#include "spirv-cross/spirv_glsl.hpp"
-#include "spirv-cross/spirv_hlsl.hpp"
-#include "spirv-cross/spirv_msl.hpp"
+#include "spirv_cross/spirv_glsl.hpp"
+#include "spirv_cross/spirv_hlsl.hpp"
+#include "spirv_cross/spirv_msl.hpp"
 #include "spirv-tools/libspirv.hpp"
 
 struct shaderc_spvc_compiler {};
