diff --git a/libshaderc_spvc/src/spvc.cc b/libshaderc_spvc/src/spvc.cc
index ccf0462..54bc7e1 100644
--- a/libshaderc_spvc/src/spvc.cc
+++ b/libshaderc_spvc/src/spvc.cc
@@ -16,9 +16,9 @@
 #include "libshaderc_util/exceptions.h"
 
 #include "spirv-tools/libspirv.hpp"
-#include "spirv-cross/spirv_glsl.hpp"
-#include "spirv-cross/spirv_hlsl.hpp"
-#include "spirv-cross/spirv_msl.hpp"
+#include "spirv_cross/spirv_glsl.hpp"
+#include "spirv_cross/spirv_hlsl.hpp"
+#include "spirv_cross/spirv_msl.hpp"
 
 struct shaderc_spvc_compiler {};
 
