--- a/gview_v4l2core/uvc_h264.c	2017-10-30 13:43:23.012551325 +0100
+++ ./gview_v4l2core/uvc_h264.c	2017-10-30 13:25:37.951372860 +0100
@@ -1039,7 +1039,7 @@
 		exit(-1);
 	}
 	
-	h264_ctx->context->flags2 |= CODEC_FLAG2_FAST;
+	h264_ctx->context->flags2 |= AV_CODEC_FLAG2_FAST;
 	h264_ctx->context->pix_fmt = AV_PIX_FMT_YUV420P;
 	h264_ctx->context->width = width;
 	h264_ctx->context->height = height;
--- a/gview_encoder/encoder.h	2017-10-30 13:42:36.724238413 +0100
+++ b/gview_encoder/encoder.h	2017-10-30 13:41:39.586320939 +0100
@@ -51,6 +51,10 @@
 #define LIBAVUTIL_VER_AT_LEAST(major,minor) 0
 #endif
 
+#ifndef X264_ME_DIA
+#define X264_ME_DIA 0
+#endif
+
 #ifndef X264_ME_HEX
 #define X264_ME_HEX 1
 #endif
--- a/gview_encoder/video_codecs.c	2017-10-30 13:42:24.944667745 +0100
+++ b/gview_encoder/video_codecs.c	2017-10-30 13:41:56.655698804 +0100
@@ -133,7 +133,7 @@
 		.codec_name   = "mjpeg",
 		.mb_decision  = 0,
 		.trellis      = 0,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 0,
@@ -169,7 +169,7 @@
 		.codec_name   = "mpeg1video",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 1,
@@ -205,11 +205,11 @@
 		.codec_name   = "flv",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 1,
-		.flags        = CODEC_FLAG_4MV
+		.flags        = AV_CODEC_FLAG_4MV
 	},
 	{
 		.valid        = 1,
@@ -241,7 +241,7 @@
 		.codec_name   = "wmv1",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 1,
@@ -277,7 +277,7 @@
 		.codec_name   = "mpeg2video",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 1,
@@ -313,7 +313,7 @@
 		.codec_name   = "msmpeg4v3",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 1,
@@ -349,7 +349,7 @@
 		.codec_name   = "mpeg4",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 1,
-		.me_method    = ME_EPZS,
+		.me_method    = X264_ME_DIA,
 		.mpeg_quant   = 1,
 		.max_b_frames = 0,
 		.num_threads  = 1,
@@ -426,7 +426,7 @@
 		.codec_name   = "libx265",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 0,
-		.me_method    = ME_HEX,
+		.me_method    = X264_ME_HEX,
 		.mpeg_quant   = 1,
 		.max_b_frames = 16,
 		.num_threads  = 4,
@@ -463,7 +463,7 @@
 		.codec_name   = "libvpx_vp8",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 0,
-		.me_method    = ME_HEX,
+		.me_method    = X264_ME_HEX,
 		.mpeg_quant   = 1,
 		.max_b_frames = 0,
 		.num_threads  = 4,
@@ -500,7 +500,7 @@
 		.codec_name   = "libvpx_vp9",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 0,
-		.me_method    = ME_HEX,
+		.me_method    = X264_ME_HEX,
 		.mpeg_quant   = 1,
 		.max_b_frames = 16,
 		.num_threads  = 4,
@@ -537,7 +537,7 @@
 		.codec_name   = "libtheora",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 0,
-		.me_method    = ME_HEX,
+		.me_method    = X264_ME_HEX,
 		.mpeg_quant   = 1,
 		.max_b_frames = 0,
 		.num_threads  = 4,
