From 947b20ee8c616d61b7eea6c911986780427819d9 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Tue, 14 Oct 2025 11:04:27 +0200
Subject: [PATCH] pcsx2 system vulkan

---
 pcsx2/CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 09c05bf6f..731bdbc5d 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -650,8 +650,7 @@ if(USE_VULKAN)
 		GS/Renderers/Vulkan/VKStreamBuffer.h
 		GS/Renderers/Vulkan/VKSwapChain.h
 	)
-	target_link_libraries(PCSX2_FLAGS INTERFACE vulkan-headers)
-	target_include_directories(PCSX2_FLAGS INTERFACE ${SHADERC_INCLUDE_DIR})
+	target_link_libraries(PCSX2_FLAGS INTERFACE Vulkan::Headers Shaderc::shaderc_shared)
 endif()
 
 set(pcsx2GSMetalShaders
-- 
2.51.0

