From b939c174b4ef0d9d10688b1621177755f90616d6 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Fri, 31 Jan 2025 15:01:48 +0100
Subject: [PATCH] fmt 11

---
 common/SmallString.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/common/SmallString.h b/common/SmallString.h
index 9359965a8..ebfbe29ef 100644
--- a/common/SmallString.h
+++ b/common/SmallString.h
@@ -378,6 +378,13 @@ __fi SmallStackString<L> SmallStackString<L>::from_vformat(fmt::string_view fmt,
 using TinyString = SmallStackString<64>;
 using SmallString = SmallStackString<256>;
 
+#if FMT_VERSION >= 100000
+template<u32 L>
+auto format_as(SmallStackString<L> s) {
+  return s.view();
+}
+#endif
+
 template <typename... T>
 __fi void SmallStringBase::append_format(fmt::format_string<T...> fmt, T&&... args)
 {
@@ -423,4 +430,4 @@ __fi void SmallStringBase::format(fmt::format_string<T...> fmt, T&&... args)
 MAKE_FORMATTER(TinyString);
 MAKE_FORMATTER(SmallString);
 
-#undef MAKE_FORMATTER
\ No newline at end of file
+#undef MAKE_FORMATTER
-- 
2.48.1

