From 3593503a054d8742f821068a35b56195e87d9339 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sun, 6 Jul 2025 09:34:41 +0200
Subject: [PATCH] system plutosvg

---
 cmake/SearchForStuff.cmake | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index ef6afde94..e360d6d2e 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -19,8 +19,6 @@ find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
 find_package(SDL3 3.2.6 REQUIRED)
 find_package(Freetype 2.11.1 REQUIRED)
-find_package(plutovg 1.1.0 REQUIRED)
-find_package(plutosvg 0.0.7 REQUIRED)
 
 if(USE_VULKAN)
 	find_package(Shaderc REQUIRED)
@@ -86,6 +84,13 @@ else()
 		find_package(RapidJSON REQUIRED)
 		add_library(rapidjson INTERFACE)
 		find_package(fmt REQUIRED)
+		pkg_check_modules(plutovg REQUIRED IMPORTED_TARGET plutovg>=1.1.0)
+		add_library(plutovg::plutovg ALIAS PkgConfig::plutovg)
+		pkg_check_modules(plutosvg REQUIRED IMPORTED_TARGET plutosvg>=0.0.7)
+		add_library(plutosvg::plutosvg ALIAS PkgConfig::plutosvg)
+	else()
+		find_package(plutovg 1.1.0 REQUIRED)
+		find_package(plutosvg 0.0.7 REQUIRED)
 	endif()
 endif()
 
-- 
2.50.0

