From f3b1ede78d0e31378e892844ade24b3150b3359f Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Wed, 30 Oct 2024 21:17:38 +0100
Subject: [PATCH] fix build

---
 pcsx2/DebugTools/SymbolGuardian.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pcsx2/DebugTools/SymbolGuardian.h b/pcsx2/DebugTools/SymbolGuardian.h
index 67bfd129c..81e2bfd71 100644
--- a/pcsx2/DebugTools/SymbolGuardian.h
+++ b/pcsx2/DebugTools/SymbolGuardian.h
@@ -4,6 +4,7 @@
 #pragma once
 
 #include <atomic>
+#include <mutex>
 #include <thread>
 #include <functional>
 #include <shared_mutex>
-- 
2.47.0

