diff --git a/pcsx2/Frontend/CommonHotkeys.cpp b/pcsx2/Frontend/CommonHotkeys.cpp
index 822471c16..77112cd48 100644
--- a/pcsx2/Frontend/CommonHotkeys.cpp
+++ b/pcsx2/Frontend/CommonHotkeys.cpp
@@ -42,7 +42,11 @@ void CommonHost::Internal::ResetVMHotkeyState()
 
 static void HotkeyAdjustTargetSpeed(double delta)
 {
+#ifdef ENABLE_ACHIEVEMENTS
 	const double min_speed = Achievements::ChallengeModeActive() ? 1.0 : 0.1;
+#else
+   const double min_speed = 0.1;
+#endif
 	EmuConfig.Framerate.NominalScalar = std::max(min_speed, EmuConfig.GS.LimitScalar + delta);
 	VMManager::SetLimiterMode(LimiterModeType::Nominal);
 	gsUpdateFrequency(EmuConfig);
@@ -253,4 +257,4 @@ DEFINE_HOTKEY_SAVESTATE_X(10)
 DEFINE_HOTKEY_LOADSTATE_X(10)
 #undef DEFINE_HOTKEY_SAVESTATE_X
 #undef DEFINE_HOTKEY_LOADSTATE_X
-END_HOTKEY_LIST()
\ No newline at end of file
+END_HOTKEY_LIST()
diff --git a/pcsx2/Frontend/FullscreenUI.cpp b/pcsx2/Frontend/FullscreenUI.cpp
index cd9642092..b1b8ab023 100644
--- a/pcsx2/Frontend/FullscreenUI.cpp
+++ b/pcsx2/Frontend/FullscreenUI.cpp
@@ -6002,14 +6002,6 @@ bool FullscreenUI::OpenLeaderboardsWindow()
 	return false;
 }
 
-void FullscreenUI::DrawAchievementsSettingsPage()
-{
-	BeginMenuButtons();
-	ActiveButton(ICON_FA_BAN "  This build was not compiled with RetroAchievements support.", false, false,
-		ImGuiFullscreen::LAYOUT_MENU_BUTTON_HEIGHT_NO_SUMMARY);
-	EndMenuButtons();
-}
-
 void FullscreenUI::DrawAchievementsLoginWindow()
 {
 }
diff --git a/pcsx2/Frontend/FullscreenUI.h b/pcsx2/Frontend/FullscreenUI.h
index 1c114da0c..097e93748 100644
--- a/pcsx2/Frontend/FullscreenUI.h
+++ b/pcsx2/Frontend/FullscreenUI.h
@@ -35,8 +35,8 @@ namespace FullscreenUI
 	void OnVMDestroyed();
 	void OnRunningGameChanged(std::string path, std::string serial, std::string title, u32 crc);
 	void OpenPauseMenu();
-	void OpenAchievementsWindow();
-	void OpenLeaderboardsWindow();
+	bool OpenAchievementsWindow();
+	bool OpenLeaderboardsWindow();
 
 	void Shutdown();
 	void Render();
