From 98cee1460315253fdd14af7349ea48a1ced43962 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Fri, 2 May 2025 12:18:43 +0200
Subject: [PATCH] system plutosvg

---
 cmake/SearchForStuff.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 840feb787..9ed43fe27 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -19,8 +19,6 @@ find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
 find_package(SDL3 3.2.6 REQUIRED)
 find_package(Freetype 2.11.1 REQUIRED)
-find_package(plutovg 0.0.13 REQUIRED)
-find_package(plutosvg 0.0.6 REQUIRED)
 
 if(USE_VULKAN)
 	find_package(Shaderc REQUIRED)
@@ -75,6 +73,10 @@ else()
 		endif()
 
 		find_package(PkgConfig REQUIRED)
+		pkg_check_modules(plutovg REQUIRED IMPORTED_TARGET plutovg>=1.0.0)
+		add_library(plutovg::plutovg ALIAS PkgConfig::plutovg)
+		pkg_check_modules(plutosvg REQUIRED IMPORTED_TARGET plutosvg>=0.0.6)
+		add_library(plutosvg::plutosvg ALIAS PkgConfig::plutosvg)
 		pkg_check_modules(DBUS REQUIRED dbus-1)
 		find_package(ryml REQUIRED)
 		add_library(rapidyaml::rapidyaml ALIAS ryml::ryml)
-- 
2.49.0

