From 6b0f20e95c473edcb5d1d7c5f11eb7bc7828309c Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Wed, 13 Aug 2025 13:45:23 +0200
Subject: [PATCH] system plutosvg

---
 cmake/SearchForStuff.cmake | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 1a80d652a..5dd566186 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -19,8 +19,6 @@ find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
 find_package(SDL3 3.2.6 REQUIRED)
 find_package(Freetype 2.12 REQUIRED)
-find_package(plutovg 1.1.0 REQUIRED)
-find_package(plutosvg 0.0.7 REQUIRED)
 
 if(USE_VULKAN)
 	find_package(Shaderc REQUIRED)
@@ -86,6 +84,13 @@ else()
 		find_package(RapidJSON REQUIRED)
 		add_library(rapidjson INTERFACE)
 		find_package(fmt REQUIRED)
+		pkg_check_modules(plutovg REQUIRED IMPORTED_TARGET plutovg>=1.1.0)
+		add_library(plutovg::plutovg ALIAS PkgConfig::plutovg)
+		pkg_check_modules(plutosvg REQUIRED IMPORTED_TARGET plutosvg>=0.0.7)
+		add_library(plutosvg::plutosvg ALIAS PkgConfig::plutosvg)
+	else()
+		find_package(plutovg 1.1.0 REQUIRED)
+		find_package(plutosvg 0.0.7 REQUIRED)
 	endif()
 endif()
 
-- 
2.50.1

