From f1a7c6eb8513837723a7082ecb7cf09b31e48d1b Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sat, 18 May 2024 15:09:12 +0200
Subject: [PATCH] fix

---
 pcsx2/CDVD/ThreadedFileReader.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcsx2/CDVD/ThreadedFileReader.cpp b/pcsx2/CDVD/ThreadedFileReader.cpp
index 3471b4f08..e301001a0 100644
--- a/pcsx2/CDVD/ThreadedFileReader.cpp
+++ b/pcsx2/CDVD/ThreadedFileReader.cpp
@@ -1,6 +1,6 @@
 // SPDX-FileCopyrightText: 2002-2023 PCSX2 Dev Team
 // SPDX-License-Identifier: LGPL-3.0+
-
+#include <string.h>
 #include "ThreadedFileReader.h"
 
 #include "common/Threading.h"
-- 
2.45.1

