diff --git a/pcsx2/PAD/Linux/SDL/joystick.cpp b/pcsx2/PAD/Linux/SDL/joystick.cpp
index 548b72e02..f20e9c4ce 100644
--- a/pcsx2/PAD/Linux/SDL/joystick.cpp
+++ b/pcsx2/PAD/Linux/SDL/joystick.cpp
@@ -48,7 +48,7 @@ void JoystickInfo::EnumerateJoysticks(std::vector<std::unique_ptr<GamePad>>& vjo
 		{ // Support as much Joystick as possible
 			GBytes* bytes = g_resource_lookup_data(PAD_res_get_resource(), "/PAD/res/game_controller_db.txt", G_RESOURCE_LOOKUP_FLAGS_NONE, nullptr);
 
-			size_t size = 0;
+			long unsigned int size = 0;
 			// SDL forget to add const for SDL_RWFromMem API...
 			void* data = const_cast<void*>(g_bytes_get_data(bytes, &size));
 
diff --git a/plugins/GSdx/GSdx.cpp b/plugins/GSdx/GSdx.cpp
index ad82cccbd..b1f43a1c3 100644
--- a/plugins/GSdx/GSdx.cpp
+++ b/plugins/GSdx/GSdx.cpp
@@ -101,7 +101,7 @@ bool GSdxApp::LoadResource(int id, std::vector<char>& buff, const char* type)
 
 	GBytes *bytes = g_resource_lookup_data(GSdx_res_get_resource(), path.c_str(), G_RESOURCE_LOOKUP_FLAGS_NONE, nullptr);
 
-	size_t size = 0;
+	long unsigned int size = 0;
 	const void* data = g_bytes_get_data(bytes, &size);
 
 	if (data == nullptr || size == 0) {
@@ -339,7 +339,7 @@ void GSdxApp::Init()
 	m_default_configuration["force_texture_clear"]                        = "0";
 	m_default_configuration["fxaa"]                                       = "0";
 	m_default_configuration["interlace"]                                  = "7";
-	m_default_configuration["conservative_framebuffer"]                   = "1"; 
+	m_default_configuration["conservative_framebuffer"]                   = "1";
 	m_default_configuration["linear_present"]                             = "1";
 	m_default_configuration["MaxAnisotropy"]                              = "0";
 	m_default_configuration["mipmap"]                                     = "1";
