diff --git a/common/src/Utilities/CMakeLists.txt b/common/src/Utilities/CMakeLists.txt
index a05ad497c..40b198575 100644
--- a/common/src/Utilities/CMakeLists.txt
+++ b/common/src/Utilities/CMakeLists.txt
@@ -102,7 +102,7 @@ set(UtilitiesFinalSources
 )
 
 foreach(library Utilities Utilities_NO_TLS)
-	add_library(${library} "${UtilitiesFinalSources}")
+	add_library(${library} STATIC "${UtilitiesFinalSources}")
 	target_link_libraries(${library} PRIVATE ${LIBC_LIBRARIES} PUBLIC wxWidgets::all)
 	target_compile_features(${library} PUBLIC cxx_std_17)
 	target_include_directories(${library} PUBLIC ../../../3rdparty/include ../../include PRIVATE ../../include/Utilities .)
