From 80191c4ecf3c7d73009744b0089632bb0866fe65 Mon Sep 17 00:00:00 2001
From: Roel Aaij <roel.aaij@gmail.com>
Date: Sun, 19 Mar 2023 14:10:40 +0100
Subject: [PATCH] visibility

---
 cmake/BuildParameters.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index 70d357ca6..c19644b76 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -151,7 +151,7 @@ option(USE_PGO_OPTIMIZE "Enable PGO optimization (use profile)")
 if(MSVC AND NOT USE_CLANG_CL)
 	add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/Zc:externConstexpr>")
 elseif(NOT MSVC)
-	add_compile_options(-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse)
+	add_compile_options(-pipe -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse)
 
 	# -fno-operator-names should only be for C++ files, not C files.
 	add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-operator-names>)
-- 
2.40.0

