diff -Naur old/src/kerberosbasic.h new/src/kerberosbasic.h
--- old/src/kerberosbasic.h	2014-03-25 10:21:31.000000000 +0100
+++ new/src/kerberosbasic.h	2014-12-06 13:19:47.187737905 +0100
@@ -15,7 +15,7 @@
  **/
 
 #include <gssapi/gssapi.h>
-#include <gssapi/gssapi_generic.h>
+#include <krb5.h>
 #include <gssapi/gssapi_krb5.h>
 
 #define krb5_get_err_text(context,code) error_message(code)
diff -Naur old/src/kerberosgss.c new/src/kerberosgss.c
--- old/src/kerberosgss.c	2015-05-17 22:35:10.000000000 +0200
+++ new/src/kerberosgss.c	2015-09-08 14:12:42.369969941 +0200
@@ -38,7 +38,8 @@
     int code;
     krb5_context kcontext;
     krb5_keytab kt = NULL;
-    krb5_kt_cursor cursor = NULL;
+    krb5_kt_cursor cursor;
+    int seq_get_started = 0;
     krb5_keytab_entry entry;
     char* pname = NULL;
     
@@ -67,6 +68,7 @@
                                                           "Cannot get sequence cursor from keytab", code));
         goto end;
     }
+    seq_get_started = 1;
     
     while ((code = krb5_kt_next_entry(kcontext, kt, &entry, &cursor)) == 0)
     {
@@ -97,7 +99,7 @@
     }
     
 end:
-    if (cursor)
+    if (seq_get_started)
         krb5_kt_end_seq_get(kcontext, kt, &cursor);
     if (kt)
         krb5_kt_close(kcontext, kt);
@@ -130,7 +132,7 @@
     if (strchr(service, '/'))
         mech = GSS_C_NO_OID;
     else
-        mech = gss_krb5_nt_service_name;
+        mech = GSS_KRB5_NT_PRINCIPAL_NAME;
 
     maj_stat = gss_import_name(&min_stat, &name_token, mech, &state->server_name);
     
diff -Naur old/src/kerberosgss.h new/src/kerberosgss.h
--- old/src/kerberosgss.h	2014-03-25 10:21:31.000000000 +0100
+++ new/src/kerberosgss.h	2014-12-06 13:21:12.507737276 +0100
@@ -15,7 +15,7 @@
  **/
 
 #include <gssapi/gssapi.h>
-#include <gssapi/gssapi_generic.h>
+#include <krb5.h>
 #include <gssapi/gssapi_krb5.h>
 
 #define krb5_get_err_text(context,code) error_message(code)
diff -Naur old/src/kerberospw.h new/src/kerberospw.h
--- old/src/kerberospw.h	2014-03-25 10:21:31.000000000 +0100
+++ new/src/kerberospw.h	2014-12-06 13:21:26.527737172 +0100
@@ -16,7 +16,7 @@
  **/
 
 #include <gssapi/gssapi.h>
-#include <gssapi/gssapi_generic.h>
+#include <krb5.h>
 #include <gssapi/gssapi_krb5.h>
 
 #define krb5_get_err_text(context,code) error_message(code)
