--- old/bindings/python/setup.py	2015-07-30 06:11:09.000000000 +0200
+++ new/bindings/python/setup.py	2015-08-14 11:50:22.508912141 +0200
@@ -71,7 +71,7 @@
 		parallel_builds = ' -j%d' % multiprocessing.cpu_count()
 
 		# build libtorrent using bjam and build the installer with distutils
-		cmdline = 'b2 boost=source link=static geoip=static boost-link=static release optimization=space stage_module --abbreviate-paths' + toolset + parallel_builds
+		cmdline = 'b2 boost=source link=dynamic geoip=dynamic boost-link=dynamic release optimization=space stage_module --abbreviate-paths' + toolset + parallel_builds
 		print(cmdline)
 		if os.system(cmdline) != 0:
 			print('build failed')
