--- a/hsacore.so.def	2024-11-25 10:54:08.018954670 +0100
+++ b/hsacore.so.def	2024-11-25 10:54:18.548890790 +0100
@@ -247,12 +247,6 @@
 	hsa_amd_vmem_retain_alloc_handle;
 	hsa_amd_vmem_get_alloc_properties_from_handle;
 	hsa_amd_agent_set_async_scratch_limit;
-	hsa_tools_scratch_event_alloc_start;
-	hsa_tools_scratch_event_alloc_end;
-	hsa_tools_scratch_event_free_start;
-	hsa_tools_scratch_event_free_end;
-	hsa_tools_scratch_event_async_reclaim_start;
-	hsa_tools_scratch_event_async_reclaim_end;
 	hsa_ven_amd_pcs_iterate_configuration;
 	hsa_ven_amd_pcs_create;
 	hsa_ven_amd_pcs_create_from_id;
