--- old/SConstruct	2017-08-02 17:30:55.929821286 +0200
+++ new/SConstruct	2017-08-02 17:30:05.569821657 +0200
@@ -427,9 +427,6 @@
     apr_libs = ''
     apu_libs = ''
 
-  env.Append(CPPPATH=['$ZLIB/include'])
-  env.Append(LIBPATH=['$ZLIB/lib'])
-
   # MacOS ships ancient OpenSSL libraries, but no headers, so we can
   # assume we're building with an OpenSSL installed outside the
   # default include and link paths. To prevent accidentally linking to
@@ -438,14 +435,9 @@
   if sys.platform == 'darwin':
     env.Prepend(CPPPATH=['$OPENSSL/include'])
     env.Prepend(LIBPATH=['$OPENSSL/lib'])
-  else:
-    env.Append(CPPPATH=['$OPENSSL/include'])
-    env.Append(LIBPATH=['$OPENSSL/lib'])
 
   if brotli:
     brotli_libs = '-lbrotlicommon -lbrotlienc'
-    env.Append(CPPPATH=['$BROTLI/include'],
-               LIBPATH=['$BROTLI/lib'])
   else:
     brotli_libs = ''
 
@@ -485,12 +481,6 @@
 if disablelogging:
     env.Append(CPPDEFINES=['SERF_DISABLE_LOGGING'])
 
-# On some systems, the -R values that APR describes never make it into actual
-# RPATH flags. We'll manually map all directories in LIBPATH into new
-# flags to set RPATH values.
-for d in env['LIBPATH']:
-  env.Append(RPATH=[':'+d])
-
 # Set up the construction of serf-*.pc
 pkgconfig = env.Textfile('serf-%d.pc' % (MAJOR,),
                          env.File('build/serf.pc.in'),
