--- a/buckets/ssl_buckets.c.orig	2018-10-25 13:09:33.486581266 +0200
+++ b/buckets/ssl_buckets.c	2018-10-25 13:09:24.486581300 +0200
@@ -299,7 +299,7 @@
 #endif

     /* The server asked to renegotiate the SSL session. */
-#ifdef TLS_ST_SW_HELLO_REQ
+#if defined(TLS_ST_SW_HELLO_REQ) || OPENSSL_VERSION_NUMBER >= 0x1010000fL
     if (SSL_get_state(s) == TLS_ST_SW_HELLO_REQ) {
 #elif defined(SSL_ST_RENEGOTIATE)
     if (SSL_state(s) == SSL_ST_RENEGOTIATE) {
@@ -1111,7 +1111,7 @@
         /* Once we got through the initial handshake, we should have received
            the ALPN information if there is such information. */
         ctx->handshake_finished = SSL_is_init_finished(ctx->ssl)
-#ifdef TLS_ST_OK
+#if defined(TLS_ST_OK) || OPENSSL_VERSION_NUMBER >= 0x1010000fL
                                   || (SSL_get_state(ctx->ssl) == TLS_ST_OK);
 #elif defined(SSL_CB_HANDSHAKE_DONE)
                                   || (SSL_state(ctx->ssl)
@@ -1419,11 +1419,6 @@
         }
 #endif
 
-#ifdef SERF_HAVE_OPENSSL_MALLOC_INIT
-        OPENSSL_malloc_init();
-#else
-        CRYPTO_malloc_init();
-#endif
         ERR_load_crypto_strings();
         SSL_load_error_strings();
         SSL_library_init();
