--- a/src/cf/osconf.m4	2019-03-08 14:17:12.389974535 +0100
+++ b/src/cf/osconf.m4	2019-03-08 14:21:48.109973483 +0100
@@ -25,9 +25,9 @@

 dnl standard programs
 AC_PROG_RANLIB
-AC_CHECK_PROGS(AS, as, [false])
+AC_CHECK_TOOL(AS, as, [false])
 AC_CHECK_PROGS(MV, mv, [false])
 AC_CHECK_PROGS(RM, rm, [false])
-AC_CHECK_PROGS(LD, ld, [false])
+AC_CHECK_TOOL(LD, ld, [false])
 AC_CHECK_PROGS(CP, cp, [false])
 AC_CHECK_PROGS(GENCAT, gencat, [false])
