--- old/configure.ac	2019-03-14 21:08:54.000000000 +0100
+++ new/configure.ac	2019-07-09 12:19:02.892190582 +0200
@@ -292,6 +292,6 @@
     save_libs=$LIBS; LIBS=$GIF_LIBS
     AC_CHECK_LIB(gif, DGifOpenFileName,
       [AC_CHECK_HEADER(gif_lib.h,
-        GIF='gif'; LIBGIF='-lgif'; gif_ok=yes,
+        GIF='gif'; LIBGIF='-lgif -lgifutil'; gif_ok=yes,
 	AC_MSG_WARN(*** GIF loader will not be built (giflibrary not found) ***))],
       AC_MSG_WARN(*** GIF loader will not be built (giflibrary not found) ***))

