--- a/setup.py	2021-07-07 17:26:36.000000000 +0200
+++ b/setup.py	2021-10-05 13:02:03.846774840 +0200
@@ -58,7 +58,7 @@
 setup(
     name="eduvpn_client",
     version=__version__,
-    packages=find_packages(exclude=['tests']),
+    packages=find_packages(exclude=['tests*']),
     data_files=data_files,
     install_requires=install_requires,
     extras_require=extras_require,
@@ -66,7 +66,6 @@
     author_email="gijs@pythonic.nl",
     description="eduVPN client",
     license="GPL3",
-    setup_requires=['pytest-runner'],
     tests_require=tests_require,
     test_suite="tests",
     keywords="vpn openvpn networking security",
