diff --git a/runners/common.go b/runners/common.go
index 11279ed4..9e8773e2 100644
--- a/runners/common.go
+++ b/runners/common.go
@@ -71,7 +71,7 @@ func locateRunnersOnce() {
 
 	paths := []string{
 		filepath.Join(filepath.Dir(exe), "llama", "build", runtime.GOOS+"-"+runtime.GOARCH, "runners"),
-		filepath.Join(filepath.Dir(exe), envconfig.LibRelativeToExe(), "lib", "ollama", "runners"),
+		filepath.Join(filepath.Dir(exe), envconfig.LibRelativeToExe(), "libexec", "ollama", "runners"),
 		filepath.Join(filepath.Dir(exe), "lib", "ollama", "runners"),
 	}
 	for _, path := range paths {
