--- a/CMakeLists.txt	2018-05-31 11:31:32.339757219 +0200
+++ b/CMakeLists.txt	2018-05-31 11:31:23.579757253 +0200
@@ -40,16 +40,6 @@
 	add_definitions(${FUSE_CFLAGS} -DFUSE_USE_VERSION=26)
 endif()
 
-check_include_files (magic.h HAVE_MAGIC_H)
-check_library_exists(magic magic_open "" HAVE_LIBMAGIC)
-
-if (HAVE_MAGIC_H AND HAVE_LIBMAGIC)
-	message(STATUS "libmagic found")
-	add_definitions(-DHAVE_LIBMAGIC)
-	list(APPEND MTP_LIBRARIES magic)
-	list(APPEND MTP_SHARED_LIBRARIES magic)
-endif()
-
 option(BUILD_QT_UI "Build reference Qt application" ON)
 option(BUILD_SHARED_LIB "Build shared library" OFF)
 
@@ -154,6 +144,16 @@
 	install(TARGETS mtp-ng-static LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
 endif (BUILD_SHARED_LIB)
 
+check_include_files (magic.h HAVE_MAGIC_H)
+check_library_exists(magic magic_open "" HAVE_LIBMAGIC)
+
+if (HAVE_MAGIC_H AND HAVE_LIBMAGIC)
+	message(STATUS "libmagic found")
+	add_definitions(-DHAVE_LIBMAGIC)
+	list(APPEND MTP_LIBRARIES magic)
+	list(APPEND MTP_SHARED_LIBRARIES magic)
+endif()
+
 add_definitions(-D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64)
 
 add_subdirectory(cli)
