diff -Naur a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,13 +22,12 @@
   "rog-anime",
   "rog-aura",
   "rog-profiles",
-  "rog-control-center",
   "rog-slash",
   "simulators",
   "rog-scsi",
 ]
 
-default-members = ["asusctl", "asusd", "asusd-user", "rog-control-center"]
+default-members = ["asusctl", "asusd", "asusd-user"]
 
 [workspace.dependencies]
 tokio = { version = "^1.39.0", default-features = false, features = [
diff -Naur a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,6 @@
 libdir = $(exec_prefix)/lib
 zshcpl = $(datarootdir)/zsh/site-functions
 
-BIN_ROG := rog-control-center
 BIN_C := asusctl
 BIN_D := asusd
 BIN_U := asusd-user
@@ -49,15 +48,11 @@
 	rm -rf .cargo vendor vendor.tar.xz
 
 install-program:
-	$(INSTALL_PROGRAM) "./target/$(TARGET)/$(BIN_ROG)" "$(DESTDIR)$(bindir)/$(BIN_ROG)"
-
 	$(INSTALL_PROGRAM) "./target/$(TARGET)/$(BIN_C)" "$(DESTDIR)$(bindir)/$(BIN_C)"
 	$(INSTALL_PROGRAM) "./target/$(TARGET)/$(BIN_D)" "$(DESTDIR)$(bindir)/$(BIN_D)"
 	$(INSTALL_PROGRAM) "./target/$(TARGET)/$(BIN_U)" "$(DESTDIR)$(bindir)/$(BIN_U)"
 
 install-data:
-	$(INSTALL_DATA) "./rog-control-center/data/$(BIN_ROG).desktop" "$(DESTDIR)$(datarootdir)/applications/$(BIN_ROG).desktop"
-	$(INSTALL_DATA) "./rog-control-center/data/$(BIN_ROG).png" "$(DESTDIR)$(datarootdir)/icons/hicolor/512x512/apps/$(BIN_ROG).png"
 	cd rog-aura/data/layouts && find . -type f -name "*.ron" -exec $(INSTALL_DATA) "{}" "$(DESTDIR)$(datarootdir)/rog-gui/layouts/{}" \;
 
 	$(INSTALL_DATA) "./data/$(BIN_D).rules" "$(DESTDIR)$(libdir)/udev/rules.d/99-$(BIN_D).rules"
@@ -86,10 +81,6 @@
 install: install-program install-data
 
 uninstall:
-	rm -f "$(DESTDIR)$(bindir)/$(BIN_ROG)"
-	rm -r "$(DESTDIR)$(datarootdir)/applications/$(BIN_ROG).desktop"
-	rm -r "$(DESTDIR)$(datarootdir)/icons/hicolor/512x512/apps/$(BIN_ROG).png"
-
 	rm -f "$(DESTDIR)$(bindir)/$(BIN_C)"
 	rm -f "$(DESTDIR)$(bindir)/$(BIN_D)"
 	rm -f "$(DESTDIR)$(libdir)/udev/rules.d/99-$(BIN_D).rules"
@@ -122,9 +113,6 @@
 	tar pcfJ vendor_asusctl_$(VERSION).tar.xz vendor
 	rm -rf vendor
 
-translate:
-	find -name \*.slint | xargs slint-tr-extractor -o rog-control-center/translations/en/rog-control-center.po
-
 build:
 ifeq ($(VENDORED),1)
 	cargo vendor
@@ -136,7 +124,6 @@
 	strip -s ./target/$(TARGET)/$(BIN_C)
 	strip -s ./target/$(TARGET)/$(BIN_D)
 	strip -s ./target/$(TARGET)/$(BIN_U)
-	strip -s ./target/$(TARGET)/$(BIN_ROG)
 endif
