--- old/tensorflow/core/platform/default/build_config/BUILD	2021-11-01 02:31:04.000000000 +0100
+++ new/tensorflow/core/platform/default/build_config/BUILD	2022-01-06 20:05:10.232057419 +0100
@@ -163,7 +163,6 @@
         "//tensorflow/core:protos_all_cc",
         "@com_googlesource_code_re2//:re2",
         "@farmhash_archive//:farmhash",
-        "@fft2d",
         "@highwayhash//:sip_hash",
         "@zlib",
     ],
--- old/third_party/fft2d/fft.h	2021-11-01 02:31:04.000000000 +0100
+++ new/third_party/fft2d/fft.h	2022-01-08 13:04:27.919813740 +0100
@@ -18,10 +18,6 @@
 #ifndef FFT2D_FFT_H__
 #define FFT2D_FFT_H__
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 extern void cdft(int, int, double *, int *, double *);
 extern void rdft(int, int, double *, int *, double *);
 extern void ddct(int, int, double *, int *, double *);
@@ -29,8 +25,4 @@
 extern void dfct(int, double *, double *, int *, double *);
 extern void dfst(int, double *, double *, int *, double *);
 
-#ifdef __cplusplus
-}
-#endif
-
 #endif  // FFT2D_FFT_H__
--- old/third_party/fft2d/fft2d.h	2021-11-01 02:31:04.000000000 +0100
+++ new/third_party/fft2d/fft2d.h	2022-01-08 13:04:36.869911351 +0100
@@ -18,10 +18,6 @@
 #ifndef FFT2D_FFT_H__
 #define FFT2D_FFT_H__
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 extern void cdft2d(int, int, int, double **, double *, int *, double *);
 extern void rdft2d(int, int, int, double **, double *, int *, double *);
 extern void ddct2d(int, int, int, double **, double *, int *, double *);
@@ -30,9 +26,4 @@
 extern void ddct8x8s(int isgn, double **a);
 extern void ddct16x16s(int isgn, double **a);
 
-
-#ifdef __cplusplus
-}
-#endif
-
 #endif  // FFT2D_FFT_H__
