--- old/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c	2016-03-03 15:48:36.000000000 +0100
+++ new/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c	2016-03-25 14:10:24.712832074 +0100
@@ -2067,8 +2067,8 @@
 	*outlen = explicitNonceLen;
     }
 
-    gcmParams.pIv = nonce;
-    gcmParams.ulIvLen = sizeof(nonce);
+    gcmParams.pNonce = nonce;
+    gcmParams.ulNonceLen = sizeof(nonce);
     gcmParams.pAAD = (unsigned char *)additionalData;  /* const cast */
     gcmParams.ulAADLen = additionalDataLen;
     gcmParams.ulTagBits = tagSize * 8;
@@ -2115,8 +2115,8 @@
     param.len = sizeof(aeadParams);
     param.data = (unsigned char *) &aeadParams;
     memset(&aeadParams, 0, sizeof(aeadParams));
-    aeadParams.pIv = (unsigned char *) additionalData;
-    aeadParams.ulIvLen = 8;
+    aeadParams.pNonce = (unsigned char *) additionalData;
+    aeadParams.ulNonceLen = 8;
     aeadParams.pAAD = (unsigned char *) additionalData;
     aeadParams.ulAADLen = additionalDataLen;
     aeadParams.ulTagLen = tagSize;
