diff --git a/CMakeLists.txt b/CMakeLists.txt
index 167a3e7..4c9d90c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,11 +52,6 @@ foreach(HEADER_FILE ${HEADER_FILES})
     install(FILES ${HEADER_FILE} DESTINATION ${HEADER_INSTALL_DIR})
 endforeach()
 
-# legacy
-add_custom_target(install-headers
-    COMMAND cmake -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/include/spirv
-        $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/include/spirv)
-
 option(SPIRV_HEADERS_SKIP_EXAMPLES "Skip building examples"
       ${SPIRV_HEADERS_SKIP_EXAMPLES})
 if(NOT ${SPIRV_HEADERS_SKIP_EXAMPLES})
