--- a/CMakeLists.txt	2018-07-03 12:58:31.074064845 +0200
+++ b/CMakeLists.txt	2018-07-03 12:58:12.174064917 +0200
@@ -52,9 +52,4 @@
     install(FILES ${HEADER_FILE} DESTINATION ${HEADER_INSTALL_DIR})
 endforeach()
 
-# legacy
-add_custom_target(install-headers
-    COMMAND cmake -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/include/spirv
-        $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/include/spirv)
-
 add_subdirectory(example)
