--- configure.ac~	2014-09-01 12:25:23.000000000 +0100
+++ configure.ac	2014-09-01 12:44:05.382612163 +0100
@@ -442,7 +442,7 @@
 if test "$enable_wayland_backend" = "yes"; then
   # For the cairo image backend
   cairo_backends="$cairo_backends cairo"
-  GDK_BACKENDS="$GDK_BACKENDS wayland"
+  GDK_BACKENDS=" wayland $GDK_BACKENDS"
   have_gio_unix=yes
   GDK_WINDOWING="$GDK_WINDOWING
 #define GDK_WINDOWING_WAYLAND"
