--- old/Imlib/load.c	2016-01-17 21:30:04.849754970 +0100
+++ new/Imlib/load.c	2016-01-17 21:26:08.739756712 +0100
@@ -454,8 +454,12 @@
   fd = fileno(f);
   /* Apparently rewind(f) isn't sufficient */
   lseek(fd, (long) 0, 0);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+  int giflib_error;
+  gif = DGifOpenFileHandle(fd, &giflib_error);
+#else
   gif = DGifOpenFileHandle(fd);
-
+#endif  
   if (!gif)
     return NULL;
   do
@@ -481,13 +485,23 @@
 	  rows = malloc(*h * sizeof(GifRowType *));
 	  if (!rows)
 	    {
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+		  int gif_errnum = 0;
+          DGifCloseFile (gif, &gif_errnum);
+#else
 	      DGifCloseFile(gif);
+#endif
 	      return NULL;
 	    }
 	  data = _imlib_malloc_image(*w, *h);
 	  if (!data)
 	    {
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+		  int gif_errnum = 0;
+          DGifCloseFile (gif, &gif_errnum);
+#else
 	      DGifCloseFile(gif);
+#endif
 	      free(rows);
 	      return NULL;
 	    }
@@ -498,7 +512,12 @@
 	      rows[i] = malloc(*w * sizeof(GifPixelType));
 	      if (!rows[i])
 		{
-		  DGifCloseFile(gif);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+		  int gif_errnum = 0;
+          DGifCloseFile (gif, &gif_errnum);
+#else
+	      DGifCloseFile(gif);
+#endif
 		  for (i = 0; i < *h; i++)
 		    if (rows[i])
 		      free(rows[i]);
@@ -587,7 +606,12 @@
 	    }
 	}
     }
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+  int gif_errnum = 0;
+  DGifCloseFile (gif, &gif_errnum);
+#else
   DGifCloseFile(gif);
+#endif
   for (i = 0; i < *h; i++)
     free(rows[i]);
   free(rows);
