--- openjdk/hotspot/make/linux/makefiles/dtrace.make	2015-03-10 23:41:04.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/dtrace.make	2015-05-03 15:39:15.025679361 +0200
@@ -32,7 +32,7 @@
 else
 
 # We need a recent GCC for the default
-ifeq "$(shell expr \( $(CC_VER_MAJOR) \>= 4 \) \& \( $(CC_VER_MINOR) \>= 4 \) )" "0"
+ifeq "$(shell expr \( \( $(CC_VER_MAJOR) \>= 4 \) \& \( $(CC_VER_MINOR) \>= 4 \) \| \( $(CC_VER_MAJOR) \>= 5 \) \) )" "0"
 REASON = "gcc version is too old"
 else
 
