--- openjdk/hotspot/src/share/vm/gc_implementation/g1/g1SATBCardTableModRefBS.cpp	2015-03-10 23:41:04.000000000 +0100
+++ openjdk/hotspot/src/share/vm/gc_implementation/g1/g1SATBCardTableModRefBS.cpp	2015-05-03 15:45:27.315676615 +0200
@@ -79,6 +79,8 @@
   }
 }
 
+template void G1SATBCardTableModRefBS::write_ref_array_pre_work<oopDesc*>(oopDesc** dst, int count);
+
 bool G1SATBCardTableModRefBS::mark_card_deferred(size_t card_index) {
   jbyte val = _byte_map[card_index];
   // It's already processed
