diff --git a/acpi_call.c b/acpi_call.c
index 3025d97..5e75b71 100644
--- a/acpi_call.c
+++ b/acpi_call.c
@@ -6,7 +6,7 @@
 #include <linux/proc_fs.h>
 #include <linux/slab.h>
 #include <asm/uaccess.h>
-#include <acpi/acpi.h>
+#include <linux/acpi.h>
 
 MODULE_LICENSE("GPL");
 
@@ -274,7 +274,7 @@ static int acpi_proc_write( struct file *filp, const char __user *buff,
         return -ENOSPC;
     }
 
-    if (copy_from_user( input, buff, len )) {
+    if (raw_copy_from_user( input, buff, len )) {
         return -EFAULT;
     }
     input[len] = '\0';
